[DM-4670] - bulk device operation change logs (#3397)
SoftwareAG/c8y-docs: content/change-logs/device-management/cumulocity-2025-186-0-bulk-device-operations-count-fix.md

---
date: 

> title: Fixed bulk device operations count

Reviewer: ```suggestion
title: Fixed multiple execution of bulk device operations
```

> title: Fixed multiple execution of bulk device operations

Reviewer: ```suggestion
title: Fixed multiple creation of bulk device operations
```

change_type:
  - value: change-VSkj2iV9m
    label: Fix
product_area: Device management & connectivity
component:
  - value: component--KIsStyzM
    label: Device Management app
build_artifact:
  - value: tc-QHwMfWtBk7
    label: cumulocity
ticket: DM-4670
version: 2025.186.0
---

> In certain scenarios, bulk device operations were executed multiple times on a single device. This issue has now been fixed.

Reviewer 2: ```suggestion
The mechanism that ensures that scheduled bulk operations are resumed after failure scenarios was unintentionally picking up ongoing bulk operations. This caused that multiple operations were created for individual devices in some scenarios. This has been solved by correcting the conditions under which a bulk operation needs to be resumed.  
```

> In specific conditions it was possible for bulk device operation to be executed more than once on a single device. This problem is now fixed.

Reviewer: ```suggestion
In certain scenarios, bulk device operations were executed multiple times on a single device. This issue has now been fixed.
```

> The mechanism that ensures that scheduled bulk operations are resumed after failure scenarios was unintentionally picking up ongoing bulk operations. This caused that multiple operations were created for individual devices in some scenarios. This has been solved by correcting the conditions under which a bulk operation needs to be resumed.  

Reviewer: ```suggestion
The mechanism that ensures that scheduled bulk operations are resumed after failure scenarios was unintentionally picking up ongoing bulk operations. This resulted in multiple operations being created for individual devices in some scenarios. This issue has been resolved by adjusting the conditions under which a bulk operation must be resumed.  
```
